core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -665,9 +665,21 @@ public static String getCarbonIndexFileName(String actualBlockName) {
    * @return
    */
   public static String getShardName(String actualBlockName) {
-    return DataFileUtil.getTaskNo(actualBlockName) + "-" + DataFileUtil.getBucketNo(actualBlockName)
-        + "-" + DataFileUtil.getSegmentNo(actualBlockName) + "-" + DataFileUtil
-        .getTimeStampFromFileName(actualBlockName);
+    String segmentNoStr = DataFileUtil.getSegmentNo(actualBlockName);
+    StringBuilder shardName = new StringBuilder();
+    // data before version 1.4 does not have segmentNo in filename
```

Comment:
Should this be moved to else block?

Reply:
And please add comment in function header to describe what is the component in shard name string